context/app/api/vitessce.py

```diff
@@ -135,21 +141,30 @@ def _build_vitessce_conf(self):
         }
 
         """
-
-        conf = ASSAY_CONF_LOOKUP[self.assay_type]["base_conf"]
+        if self.assay_type not in ASSAY_CONF_LOOKUP:
+            return {}
         files = ASSAY_CONF_LOOKUP[self.assay_type]["files_conf"]
-        layers = [
-            self._build_layer_conf(file, self.uuid, self.assay_type) for file in files
-        ]
+        file_paths_expected = [file["rel_path"] for file in files]
+        file_paths_found = [file["rel_path"] for file in self.entity["files"]]
+        # Codex assay needs to be built up.
+        # We need a better way to handle it but for now, this is ok.
+        if self.assay_type != CODEX_ASSAY:
+            # We need to check that the files we expect actually exist.
+            # This is due to the volatility of the datasets.
+            if not set(file_paths_expected).issubset(set(file_paths_found)):
+                return {}
```

Comment:
Maybe a warning to the log? I'm not sure where the boundary is between a negative that is entirely normal, and an expected state that should not be ignored silently.